src/hjson_value.cpp

```diff
@@ -347,6 +347,27 @@ Value& Value::at(const char *name) {
   return at(std::string(name));
 }
 
+#if __cplusplus >= 201703L
+void Value::insert(const std::string&key, Value&& other)
+{
+  switch (prv->type)
+  {
+  case Type::Undefined:
+    prv->~ValueImpl();
+    // Recreate the private object using the same memory block.
+    new(&(*prv)) ValueImpl(Type::Map);
+    // Fall through
+  case Type::Map:
+    try {
+      prv->m->m.insert_or_assign(key, std::move(other));
+    } catch(const std::out_of_range&) {}    
```

Comment:
What would cause `std::out_of_range` to be thrown, and why should it be ignored?

Reply:
You are right in that it does not make a lot of sense, maybe just not have a catch and throw whatever exception came from the container?

Reply:
Yes better to not have a catch.